Doc/library/fractions.rst

```diff
@@ -114,29 +114,15 @@ another rational number, or from a string.
 
       .. versionadded:: 3.8
 
-   .. method:: from_float(flt)
+   .. classmethod:: from_float(flt)
 
-      This class method constructs a :class:`Fraction` representing the exact
-      value of *flt*, which must be a :class:`float`. Beware that
-      ``Fraction.from_float(0.3)`` is not the same value as ``Fraction(3, 10)``.
+      Alternative constructor which only accepts instances of
+      :class:`float` or :class:`numbers.Integral`.
 
-      .. note::
-
-         From Python 3.2 onwards, you can also construct a
-         :class:`Fraction` instance directly from a :class:`float`.
-
-
-   .. method:: from_decimal(dec)
-
-      This class method constructs a :class:`Fraction` representing the exact
-      value of *dec*, which must be a :class:`decimal.Decimal` instance.
-
-      .. note::
-
-         From Python 3.2 onwards, you can also construct a
```

Comment:
Like decimal.from_float, we need to keep some note this functionality was mostly supplanted.

Reply:
Mostly, I don't think this edit improves the docs.  It is okay to make the from_float/decimal sections more brief (and to fix their markup) but we need to keep some explanation that their functionality has mostly be supplanted by the main constructors.